batchflow/research/workers.py

```diff
@@ -107,14 +107,15 @@ def __call__(self, queue, results):
                                 signal = feedback_queue.get(timeout=1)
                             except EmptyException:
                                 signal = None
-                            if signal is None and (time.time() - last_update_time.value) / 60 > self.timeout:
-                                p = psutil.Process(pid)
-                                p.terminate()
-                                message = 'Job {} [{}] failed in {}'.format(job[0], pid, self.worker_name)
-                                self.logger.info(message)
-                                final_signal.exception = TimeoutError(message)
-                                results.put(copy(final_signal))
-                                break
+                            if signal is None:
+                                if self.timeout is not None and (time.time() - last_update_time.value) / 60 > self.timeout:
+                                    p = psutil.Process(pid)
+                                    p.terminate()
+                                    message = 'Job {} [{}] failed in {}'.format(job[0], pid, self.worker_name)
```

Comment:
Please, change the message to something more descriptive like `failed because of timeout`